Source/MLX/MLXArray+Indexing.swift

```diff
@@ -158,212 +153,46 @@ extension MLXArray {
 
             precondition(dtype == newValue.dtype, "\(dtype) != \(newValue.dtype)")
 
-            // trim off any singleton dimensions
-            let updateShape = Array(newValue.shape.trimmingPrefix { $0 == 1 })
-
             // this is the shape we will set via broadcast.
             let axis = MLX.resolve(axis: axis, ndim: ndim)
             var broadcastShape = self.shape
             broadcastShape[axis] = 1
 
-            let expanded = newValue.reshaped(updateShape).broadcast(to: broadcastShape.asInt32)
+            let expanded =
+                newValue
+                .leadingSingletonDimensionsRemoved(stream: stream)
+                .broadcast(to: broadcastShape.asInt32)
 
             let indices = [resolve(index: index, axis: axis)]
             self.update(scattered(indices: indices, updates: expanded, axes: [axis.int32]))
         }
     }
 
-    @inlinable
-    public subscript(indices: Int..., stream stream: StreamOrDevice = .default) -> MLXArray {
-        get {
-            self[indices, stream: stream]
-        }
-        set {
-            self[indices, stream: stream] = newValue
-        }
-    }
-
-    /// Multiple axis subscript.
+    /// Indexing using a range expression (e.g. `0 ..< 5`) with an axis (deprecated).
+    ///
+    /// This method supports a range expression with an axis, for example this python code:
+    ///
+    /// ```python
+    /// # python
+    /// array[..., 0:3]
+    /// ```
+    ///
+    /// This can be translated to swift as:
     ///
     /// ```swift
-    /// // print the contents of [1, 3]
-    /// print(array[1, 3])
+    /// array[.ellipsis, 0 ..< 3]
+    /// ```
     ///
-    /// // set the value of [0, 7] to otherArray (with broadcast)
-    /// array[0, 7] = otherArray
+    /// This particular subscript operator is deprecated in favor of that, but for
+    /// completeness the equivalent would be:
+    ///
+    /// ```swift
+    /// array[0 ..< 3, axis: -1]
     /// ```
-    public subscript(indices: [Int], stream stream: StreamOrDevice = .default) -> MLXArray {
```

Comment:
These are subsumed by the new API and can be deleted without breaking _source_ compatibility.